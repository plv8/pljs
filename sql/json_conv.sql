CREATE FUNCTION conv(o json) RETURNS json AS $$
if (o instanceof Array) {
	o[1] = 10;
} else if (typeof(o) == 'object') {
	o.i = 10;
}
return o;
$$ LANGUAGE pljs;

SELECT conv('{"i": 3, "b": 20}');
SELECT conv('[1, 2, 3]');
