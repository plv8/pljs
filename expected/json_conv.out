CREATE FUNCTION conv(o json) RETURNS json AS $$
if (o instanceof Array) {
	o[1] = 10;
} else if (typeof(o) == 'object') {
	o.i = 10;
}
return o;
$$ LANGUAGE pljs;
SELECT conv('{"i": 3, "b": 20}');
      conv       
-----------------
 {"i":10,"b":20}
(1 row)

SELECT conv('[1, 2, 3]');
   conv   
----------
 [1,10,3]
(1 row)

